### PR TITLE
Direction 2: Law & regs tooling

### PR DESCRIPTION
**Pull request by kewitham**
This PR adds in page search functionality to the law and regs side nav.
Future PRs will build the mobile view and the sticky view.

Image

### CODE REVIEW

**Comment by kewitham** [comment]
> (a11y tests are failing)

it looks like this is related to the copy/print/share buttons (seems to fail intermittently for these)

File: assets/js/laws-regs-parser.js [hunk 1]
```diff
@@ -242,3 +235,157 @@ function shareLink(e, divType) {
...
+    if (newCount <= 0) {
+      prevBtn.classList.add('disabled');
+    }
+    if (newCount < totalCount) {
+      nextBtn.classList.remove('disabled');
+    }
+  }
+}
+
+function search() {
```
Comment (okkays) [inline assets/js/laws-regs-parser.js:299]: this function is pretty large

I feel like there are a couple of different things going on here (highlighting, dealing with buttons)

Splitting those off might make it a bit more readable

File: assets/js/laws-regs-parser.js [hunk 2]
```diff
@@ -242,3 +235,157 @@ function shareLink(e, divType) {
...
+      section.classList.remove('searched');
+      const detailEls = section.querySelectorAll('details');
+      detailEls.forEach((detailEl) => detailEl.removeAttribute('open'));
+    }
+  });
+  const results = document.querySelectorAll('.search-term');
+  if (results.length > 0) {
+    results.forEach((result, i) => {
+      const count = i + 1;
+      result.id = 'result' + count;
```
Comment (okkays) [inline assets/js/laws-regs-parser.js:330]: I worry about calling this `result` - there's a good chance someone adds some other `result` in the future that conflicts.

Does this have to be an ID, or can you make it a class and use

```
someContainer.querySelector(`.result-${i}`)
```

to find it instead?

**Review by okkays — CHANGES_REQUESTED** [review]

File: assets/js/laws-regs-parser.js (hunk 2 above)
Comment (kewitham) [reply assets/js/laws-regs-parser.js]: I used id so I can use location.hash to take the user to the search term

**Review by okkays — APPROVED** [review]
LGTM!

**Review by vwilliamsn — APPROVED** [review]
LGTM, thanks @kewitham!

**Pull request closed without merging**